src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal static class GeneratedCodeUtilities
+    {
+        private static readonly string[] s_autoGeneratedStrings = new[] { "<autogenerated", "<auto-generated" };
+
+        internal static bool IsGeneratedSymbolWithGeneratedCodeAttribute(
```

Comment:
Tagging @mattwar @srivatsn @gafter as I recall the design team had reservations about defining a common "GeneratedCodeUtilities" type that is shared across all Roslyn components (analyzer driver, IDE, etc.) as definition of what classifies as Generated code might differ across components.